2025.05.01 breaks with certain compatible versions of `attrs` installed

### Current Behavior

Receive the following error when invoking nox 2025.05.01: `AttributeError: module 'attrs.validators' has no attribute 'or_'` 
`attrs==23.1` is installed, which is compatible with nox dependency `>=23.1`
This is because as of [PR 932](https://github.com/wntrblm/nox/pull/932) nox uses `.or_` which was not introduced til attrs v24.1.0

![Image](https://github.com/user-attachments/assets/a7a0e696-f64b-45b3-a6df-d9731fb50f6c)



### Expected Behavior

Nox can be invoked with all versions consistent with dependency >=23.1 (including attrs < 24.1 which is affected by this issue), or bump attrs dependency

### Steps To Reproduce
Install attrs 23.1
Install nox 2025.05.01
Run nox

I have this same error.

```sh
poetry run nox --error-on-external-run --reuse-venv=yes --non-interactive
Traceback (most recent call last):
  File "/home/runner/.cache/pypoetry/virtualenvs/aws-lambda-powertools-2Yi3cwHL-py3.11/bin/nox", line 5, in <module>
    from nox.__main__ import main
  File "/home/runner/.cache/pypoetry/virtualenvs/aws-lambda-powertools-2Yi3cwHL-py3.11/lib/python3.11/site-packages/nox/__init__.py", line 20, in <module>
    from nox._cli import main
  File "/home/runner/.cache/pypoetry/virtualenvs/aws-lambda-powertools-2Yi3cwHL-py3.11/lib/python3.11/site-packages/nox/_cli.py", line 33, in <module>
    from nox import _options, tasks, workflow
  File "/home/runner/.cache/pypoetry/virtualenvs/aws-lambda-powertools-2Yi3cwHL-py3.11/lib/python3.11/site-packages/nox/_options.py", line 28, in <module>
    from nox import _option_set
  File "/home/runner/.cache/pypoetry/virtualenvs/aws-lambda-powertools-2Yi3cwHL-py3.11/lib/python3.11/site-packages/nox/_option_set.py", line 51, in <module>
    av_opt_path = av.optional(av.or_(av.instance_of(str), av.instance_of(os.PathLike)))
                              ^^^^^^
AttributeError: module 'attrs.validators' has no attribute 'or_'
make: *** [Makefile:48: test-dependencies] Error 1
Error: Process completed with exit code 2.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -62,6 +62,8 @@ jobs:
         run:  uv pip install --system .
       - name: Run tests on ${{ matrix.os }}
         run: nox --session "tests-${{ matrix.python-version }}" -- --full-trace
+      - name: Run min-version tests on ${{ matrix.os }}
+        run: nox --session "tests-${{ matrix.python-version }}" -- --full-trace
       - name: Run Conda tests
         if: matrix.python-version == '3.12'
         run: nox --session "conda_tests" -- --full-trace

--- a/noxfile.py
+++ b/noxfile.py
@@ -67,7 +67,17 @@ def tests(session: nox.Session) -> None:
             con.execute("DELETE FROM file WHERE SUBSTR(path, 2, 1) == ':'")
 
 
-@nox.session(venv_backend="conda", default=shutil.which("conda"))
+@nox.session(venv_backend="uv", default=False)
+def minimums(session: nox.Session) -> None:
+    """Run test suite with the lowest supported versions of everything. Requires uv."""
+    session.create_tmp()
+
+    session.install("-e.", "--group=test", "--resolution=lowest-direct")
+    session.run("uv", "pip", "list")
+    session.run("pytest", *session.posargs)
+
+
+@nox.session(venv_backend="conda", default=bool(shutil.which("conda")))
 def conda_tests(session: nox.Session) -> None:
     """Run test suite set up with conda."""
     session.conda_install(

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -39,12 +39,14 @@ classifiers = [
 ]
 dependencies = [
   "argcomplete>=1.9.4,<4",
-  "attrs>=23.1",
+  "attrs>=24.1",
   "colorlog>=2.6.1,<7",
   "dependency-groups>=1.1",
-  "packaging>=20.9",
-  "tomli>=1; python_version<'3.11'",
-  "virtualenv>=20.14.1",
+  "packaging>=20.9; python_version<'3.10'",
+  "packaging>=21; python_version>='3.10'",
+  "tomli>=1.1; python_version<'3.11'",
+  "virtualenv>=20.14.1; python_version<'3.10'",
+  "virtualenv>=20.15; python_version>='3.10'",
 ]
 optional-dependencies.tox_to_nox = [
   "importlib-resources; python_version<'3.9'",
@@ -69,7 +71,7 @@ dev = [
 test = [
   "coverage[toml]>=7.2",
   "pytest>=6",
-  "pytest-cov",
+  "pytest-cov>=3",
 ]
 docs = [
   "myst-parser",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -70,7 +70,8 @@ dev = [
 ]
 test = [
   "coverage[toml]>=7.2",
-  "pytest>=6",
+  "pytest>=7.4; python_version>='3.12'",
+  "pytest>=7; python_version<'3.12'",
   "pytest-cov>=3",
 ]
 docs = [

--- a/tests/test_virtualenv.py
+++ b/tests/test_virtualenv.py
@@ -1015,9 +1015,16 @@ def test_inner_functions_reusing_venv(
     venv, location = make_one(reuse_existing=True)
     venv.create()
 
+    txt = location.joinpath("pyvenv.cfg").read_text(encoding="utf-8")
+    cfg = {
+        (v := line.partition("="))[0].strip(): v[-1].strip()
+        for line in txt.splitlines()
+    }
+    home = cfg["home"]
+
     # Drop a venv-style pyvenv.cfg into the environment.
-    pyvenv_cfg = """\
-    home = /usr/bin
+    pyvenv_cfg = f"""\
+    home = {home}
     include-system-site-packages = false
     version = 3.10
     base-prefix = foo

